### PR TITLE
astradb[minor]: update dependencies for compatibility with langchain-core 0.3

### PR DESCRIPTION
**Pull request by ccurme**

### CODE REVIEW

**Comment by ccurme** [comment]
Integration tests may be failing for unrelated reasons. See [this run](https://github.com/langchain-ai/langchain-datastax/actions/runs/10889850276/job/30217354727?pr=67) on a recently merged PR that failed the same way.

**Comment by cbornet** [comment]
> Integration tests may be failing for unrelated reasons

Yes integration tests have setup issues.

**Comment by cbornet** [comment]
I think we should follow LangChain baseline and drop support for the EOL'd Python 3.8.
cc @hemidactylus

**Comment by ccurme** [comment]
Updated!

**Review by cbornet — APPROVED** [review]

File: libs/astradb/pyproject.toml [hunk 1]
```diff
@@ -11,8 +11,8 @@ license = "MIT"
 "Source Code" = "https://github.com/langchain-ai/langchain-datastax/tree/main/libs/astradb"
 
 [tool.poetry.dependencies]
-python = ">=3.8.1,<4.0"
-langchain-core = ">=0.1.31,<0.3"
+python = ">=3.9,<4.0"
+langchain-core = ">=0.1.31,<0.4"
 astrapy = "^1.4"
 # Resolve numpy version for 3.8 to 3.12+
```
Comment (cbornet) [inline libs/astradb/pyproject.toml:17]: The numpy resolution can probably be removed now to only use > 1.26

Comment (ccurme) [reply libs/astradb/pyproject.toml:17]: Good call thanks.

**Comment by ccurme** [comment]
> > Integration tests may be failing for unrelated reasons
> 
> Yes integration tests have setup issues.

@cbornet @hemidactylus the release pipeline appears [blocked](https://github.com/langchain-ai/langchain-datastax/actions/runs/10948233310) by failing integration tests. I suspect they are unrelated to this change and related to test infra. Would you mind taking a look?

**Comment by hemidactylus** [comment]
> > > Integration tests may be failing for unrelated reasons
> > 
> > 
> > Yes integration tests have setup issues.
> 
> @cbornet @hemidactylus the release pipeline appears [blocked](https://github.com/langchain-ai/langchain-datastax/actions/runs/10948233310) by failing integration tests. I suspect they are unrelated to this change and related to test infra. Would you mind taking a look?

@ccurme Sure, the state of CI in this repo is on my TODOs for today. Thanks for reminding me :)

**Comment by hemidactylus** [comment]
@ccurme FYI #82 should pass all tests and get into main momentarily (if all goes well)

**Pull request merged**